eth/stagedsync/stage_headers.go

```diff
@@ -397,10 +402,10 @@ func HeadersUnwind(u *UnwindState, s *StageState, tx kv.RwTx, cfg HeadersCfg, te
 			return fmt.Errorf("iterate over headers to mark bad headers: %w", err)
 		}
 	}
-	if err := rawdb.TruncateCanonicalHash(tx, u.UnwindPoint+1, badBlock); err != nil {
+	if err := rawdb.TruncateCanonicalHash(tx, u.UnwindPoint+1, u.Reason.IsBadBlock()); err != nil {
```

Comment:
I think we should pass explicit true here because even in case of non-bad blocks we don't want that they can be queried by the number if chain has already been rewinded.